java/src/main/native/src/ColumnVectorJni.cpp

```diff
@@ -299,4 +299,23 @@ JNIEXPORT jlongArray JNICALL Java_ai_rapids_cudf_ColumnVector_cudfSlice(JNIEnv *
   }
   CATCH_STD(env, NULL);
 }
+
+JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnVector_cudfLengths(JNIEnv *env, jclass clazz,
+                                                                     jlong column_handle) {
+  JNI_NULL_CHECK(env, column_handle, "input column is null", 0);
+  try {
+    gdf_column *n_column = reinterpret_cast<gdf_column *>(column_handle);
+    NVStrings *strings = static_cast<NVStrings *>(n_column->data);
+    cudf::jni::gdf_column_wrapper lengths(n_column->size, gdf_dtype::GDF_INT32,
```

Comment:
I'd like to see a check here that `n_column->size == strings->size()` and throw IllegalArgumentException or AssertionError if not true. The code critically depends upon this. If somehow the gdf_column and NVStrings sizes ever got out of sync then this code will either corrupt GPU memory or return a column with uninitialized values, both of which are difficult to debug.

Reply:
PTAL